openproblems/tasks/_cell_cell_communication/_common/methods/baseline.py

```diff
@@ -0,0 +1,43 @@
+from .....tools.decorators import method
+from .....tools.utils import check_version
+
+import numpy as np
+import pandas as pd
+
+
+@method(
+    method_name="Random Events",
+    paper_name="Random Events (baseline)",
+    paper_url="https://openproblems.bio",
+    paper_year=2022,
+    code_url="https://github.com/openproblems-bio/openproblems",
+    is_baseline=True,
+)
+def random_events(adata, test=False, n_events=1000):
+    adata.uns["ccc_pred"] = pd.DataFrame(
+        {
+            "ligand": np.random.choice(adata.var.index, n_events),
+            "receptor": np.random.choice(adata.var.index, n_events),
+            "source": np.random.choice(adata.obs["label"].cat.categories, n_events),
+            "target": np.random.choice(adata.obs["label"].cat.categories, n_events),
+            "score": np.random.uniform(0, 1, n_events),
+        }
+    )
+    adata.uns["method_code_version"] = check_version("openproblems")
+    return adata
+
+
+@method(
+    method_name="True Events",
+    paper_name="True Events (baseline)",
+    paper_url="https://openproblems.bio",
+    paper_year=2022,
+    code_url="https://github.com/openproblems-bio/openproblems",
+    is_baseline=True,
+)
+def true_events(adata, test=False):
+    adata.uns["ccc_pred"] = adata.uns["ccc_target"].rename(
+        {"response": "score"}, axis=1
+    )
+    adata.uns["method_code_version"] = check_version("openproblems")
+    return adata
```

Comment:
This one looks good. Would only need to add `.sort_values('score', ascending=False)` so that 1s are always at the top for the `odds_ratios`.

In the PR, I have now requested every method to identify whether the score predictions should be ascending for the odds ratios, i.e. a function that passes the tests should look like this:

```
def true_events(adata, test=False, ascending=False):
    adata.uns["ccc_pred"] =  (
        adata.uns["ccc_target"]
        .rename({"response": "score"}, axis=1)
        .sort_values('score', ascending=False)
    )
    
    adata.uns['ascending'] = ascending
    adata.uns["method_code_version"] = check_version("openproblems")
    return adata
```

Reply:
What if we just either a) force the scores to be ascending or b) require the metrics sort the scores before computation if necessary?

Reply:
Some should be descending, so option b is viable and is already the case actually, as liana does sort them.

 I just realized that perhaps it's not explicit enough when I saw the perfect baseline and thought to make it required. I'm open to either solution.

Reply:
Solved by #629